sentry-ruby/lib/sentry/client.rb

```diff
@@ -25,6 +25,7 @@ def initialize(configuration)
 
     def capture_event(event, scope, hint = {})
       return unless configuration.sending_allowed?
+      return unless configuration.sample_allowed? || event.is_a?(TransactionEvent)
```

Comment:
maybe the other way round so we short circuit the rng call completely if it's a transaction?
```suggestion
      return unless event.is_a?(TransactionEvent) || configuration.sample_allowed?
```